File: CPP/Menu.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

class menu {
    // Atribut bersifat rahasia
    private:
        vector<Produk> daftarProduk;

    // Method yang bersifat public
    public:

        void tambahProduk() {
            // Membuat atribut untuk penyimpanan sementara
            string id, nama, harga, stok;

            // Membuat fungsi pemasukan data
            cout << "Masukan Id Produk:" << endl; cin >> id;

            cout << "Masukan Nama Produk:" << endl; cin >> nama;
            
            cout << "Masukan Harga Produk:" << endl; cin >> harga;
            
            cout << "Masukan Stok Produk:" << endl; cin >> stok;
        
            // Membuat objek dengan constructor
            Produk p(id, nama, harga, stok);

            // Memasukan hasil kedalam vector
            daftarProduk.push_back(p);
        }

        void tampilProduk() { 
            // Buat perulangan untuk membaca setiap data dari daftar produk
            for (int i = 0; i < daftarProduk.size(); i++) {
                cout << "----------------------" << endl;
                cout << daftarProduk[i].getId() << endl;
                cout << daftarProduk[i].getNama() << endl;
                cout << daftarProduk[i].getHarga() << endl;
                cout << daftarProduk[i].getStok() << endl;
            }
            cout << "----------------------" << endl;
        }

        void updateData() {
            // Membuat atribut penyimpan sementara
            string id_target;

            // Membuat konfismasi pemasukan id produk yang akan di ubah
            cout << "Masukan id dari produk :" << endl; cin >> id_target;

            // Membuat tanda untuk hasil pencarian
            bool found = false;

            // Memulai operasi pencarian
            for (int i = 0; i < daftarProduk.size(); i++) {
                // Jika id target ditemukan
                if (daftarProduk[i].getId() == id_target) {
                    // Ubah tanda menjadi true
                    found = true;

                    // Tampilkan data lama untuk perbandingan
                    cout << "----------------------" << endl;
                    cout << "Data lama:" << endl;
                    cout << daftarProduk[i].getId() << endl;
                    cout << daftarProduk[i].getNama() << endl;
                    cout << daftarProduk[i].getHarga() << endl;
                    cout << daftarProduk[i].getStok() << endl;
                    cout << "----------------------" << endl;

                    // Membuat konfirmasi untuk memasukan data baru
                    cout << "Masukan data baru" << endl;
                    string namaBaru, hargaBaru, stokBaru;
                    
                    // Pengisian data baru
                    cout << "Nama baru :" << endl; cin >> namaBaru;
                    cout << "harga baru :" << endl; cin >> hargaBaru;
                    cout << "stok baru :" << endl; cin >> stokBaru;

                    // Proses pengubahan data
                    daftarProduk[i].setNama(namaBaru);
                    daftarProduk[i].setHarga(hargaBaru);
                    daftarProduk[i].setStok(stokBaru);

                    // Konfirmasi bahwa data berhasil di update!!
                    cout << "----------------------" << endl;
                    cout << "Data berhasil diupdate!" << endl;
                    cout << "----------------------" << endl;
                }
            }

            // Kondisi jika data tidak ditemukan
            if (found == false) {
                // Konfirmasi bahwa data tidak di temukan
                cout << "Produk dengan ID tersebut tidak ditemukan" << endl;
            }
        }

        void hapusData() {
            // Membuat atribut penyimpan sementara
            string id_target;

            // Membuat konfismasi pemasukan id produk yang akan di ubah
            cout << "Masukan id dari produk :" << endl; cin >> id_target;

            // Membuat tanda untuk hasil pencarian
            bool found = false;

            // Membuat tanda untuk iterasi
            int i = 0;
            // Memulai operasi pencarian
            while (found == false && i <= daftarProduk.size()) {
                // Jika id target ditemukan
                if (daftarProduk[i].getId() == id_target) {
                    // Ubah tanda menjadi true
                    found = true;
    
                    // Tampilkan data yang akan di hapus
                    cout << "----------------------" << endl;
                    cout << "Data yang akan dihapus:" << endl;
                    cout << daftarProduk[i].getId() << endl;
                    cout << daftarProduk[i].getNama() << endl;
                    cout << daftarProduk[i].getHarga() << endl;
                    cout << daftarProduk[i].getStok() << endl;
                    cout << "----------------------" << endl;
    
                    // Memulai penghapusan data
                    daftarProduk.erase(daftarProduk.begin() + i);
    
                    // Konfirmasi bahwa data telah dihapus
                    cout << "Produk dengan ID tersebut berhasil terhapus!" << endl;
                }
                // iterasikan tanda penghitung
                i++;
            }
            
            // Kondisi jika data tidak ditemukan
            if (found == false) {
                // Konfirmasi bahwa data tidak di temukan
                cout << "Produk dengan ID tersebut tidak ditemukan" << endl;
            }
        }

        void cariData() {
            // Membuat atribut penyimpan sementara
            string id_target;

            // Membuat konfismasi pemasukan id produk yang akan di ubah
            cout << "Masukan id dari produk :" << endl; cin >> id_target;

            // Membuat tanda untuk hasil pencarian
            bool found = false;

            // Memulai operasi pencarian
            for(int i = 0; i < daftarProduk.size(); i++) {
                // Jika id target ditemukan
                if (daftarProduk[i].getId() == id_target) {
                    // Ubah tanda menjadi true
                    found = true;

                    // Tampilkan data 
                    cout << "----------------------" << endl;
                    cout << "Produk ditemukan :" << endl;
                    cout << "----------------------" << endl;
                    cout << daftarProduk[i].getId() << endl;
                    cout << daftarProduk[i].getNama() << endl;
                    cout << daftarProduk[i].getHarga() << endl;
                    cout << daftarProduk[i].getStok() << endl;
                    cout << "----------------------" << endl;
                }
            }
            // Kondisi jika data tidak ditemukan
            if (found == false) {
                // Konfirmasi bahwa data tidak di temukan
                cout << "Produk dengan ID tersebut tidak ditemukan" << endl;
            }
        }
};

File: JAVA/Menu.java
import java.util.ArrayList;
import java.util.Scanner;

public class Menu {
    private ArrayList<Produk> daftarProduk;

    public Menu() {
        daftarProduk = new ArrayList<>();
    }

    public void tambahProduk() {
        Scanner sc = new Scanner(System.in);

        System.out.println("Masukan Id Produk:");
        String id = sc.nextLine();

        System.out.println("Masukan Nama Produk:");
        String nama = sc.nextLine();

        System.out.println("Masukan Harga Produk:");
        String harga = sc.nextLine();

        System.out.println("Masukan Stok Produk:");
        String stok = sc.nextLine();

        Produk p = new Produk(id, nama, harga, stok);
        daftarProduk.add(p);
    }

    public void tampilProduk() {
        for (Produk p : daftarProduk) {
            System.out.println("----------------------");
            System.out.println(p.getId());
            System.out.println(p.getNama());
            System.out.println(p.getHarga());
            System.out.println(p.getStok());
        }
        System.out.println("----------------------");
    }

    public void updateData() {
        Scanner sc = new Scanner(System.in);
        System.out.println("Masukan id dari produk:");
        String id_target = sc.nextLine();

        boolean found = false;

        for (Produk p : daftarProduk) {
            if (p.getId().equals(id_target)) {
                found = true;

                System.out.println("----------------------");
                System.out.println("Data lama:");
                System.out.println(p.getId());
                System.out.println(p.getNama());
                System.out.println(p.getHarga());
                System.out.println(p.getStok());
                System.out.println("----------------------");

                System.out.println("Masukan data baru");
                System.out.print("Nama baru: ");
                String namaBaru = sc.nextLine();
                System.out.print("Harga baru: ");
                String hargaBaru = sc.nextLine();
                System.out.print("Stok baru: ");
                String stokBaru = sc.nextLine();

                p.setNama(namaBaru);
                p.setHarga(hargaBaru);
                p.setStok(stokBaru);

                System.out.println("----------------------");
                System.out.println("Data berhasil diupdate!");
                System.out.println("----------------------");
                break;
            }
        }

        if (!found) {
            System.out.println("Produk dengan ID tersebut tidak ditemukan");
        }
    }

    public void hapusData() {
        Scanner sc = new Scanner(System.in);
        System.out.println("Masukan id dari produk:");
        String id_target = sc.nextLine();

        boolean found = false;

        for (int i = 0; i < daftarProduk.size(); i++) {
            Produk p = daftarProduk.get(i);
            if (p.getId().equals(id_target)) {
                found = true;

                System.out.println("----------------------");
                System.out.println("Data yang akan dihapus:");
                System.out.println(p.getId());
                System.out.println(p.getNama());
                System.out.println(p.getHarga());
                System.out.println(p.getStok());
                System.out.println("----------------------");

                daftarProduk.remove(i);
                System.out.println("Produk dengan ID tersebut berhasil terhapus!");
                break;
            }
        }

        if (!found) {
            System.out.println("Produk dengan ID tersebut tidak ditemukan");
        }
    }

    public void cariData() {
        Scanner sc = new Scanner(System.in);
        System.out.println("Masukan id dari produk:");
        String id_target = sc.nextLine();

        boolean found = false;

        for (Produk p : daftarProduk) {
            if (p.getId().equals(id_target)) {
                found = true;

                System.out.println("----------------------");
                System.out.println("Produk ditemukan:");
                System.out.println("----------------------");
                System.out.println(p.getId());
                System.out.println(p.getNama());
                System.out.println(p.getHarga());
                System.out.println(p.getStok());
                System.out.println("----------------------");
                break;
            }
        }

        if (!found) {
            System.out.println("Produk dengan ID tersebut tidak ditemukan");
        }
    }
}